### PR TITLE
updated PZ21

### PR DESCRIPTION
**Pull request by kate-bobryk**
Please check updated PZ21.

### CODE REVIEW

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  //проверить что видна форма для авторизации;
+  I.waitForVisible(loginPage.loginForm, 5); //лучше использовать для подвисаний (по умолчанию сек)
+  //проверить что в CS 1 подпункт, а в SS 6;
+  await loginPage.assertNumbersOfVisibleElements(loginPage.clientSideLoginContainer, 1);
+  await loginPage.assertNumbersOfVisibleElements(loginPage.serverSideLoginContainer, 6);
+  //проверить что все названия вкладок в CS и SS отображаются правильно (нет ошибок типа EvEnTStrEAm и т.д.);
+  //(проверить проход по масиву server side c проверкой по циклам по каждому элемента);
+  const serverSide = await I.grabTextFromAll(loginPage.serverSideLoginContainer);
+  loginPage.checkElements(serverSide, false);
+  const clientSide = await I.grabAttributeFrom(loginPage.clientSideLoginContainer, "title"); //проверить валидный ли локатор!
```
Comment (KaledinaDARIA) [inline CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts:20]: remove comment

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  //открыть страницу логина;
+  I.amOnPage("");
+  //проверить что видна форма для авторизации;
+  I.waitForVisible(loginPage.loginForm, 5); //лучше использовать для подвисаний (по умолчанию сек)
+  //проверить что в CS 1 подпункт, а в SS 6;
+  await loginPage.assertNumbersOfVisibleElements(loginPage.clientSideLoginContainer, 1);
+  await loginPage.assertNumbersOfVisibleElements(loginPage.serverSideLoginContainer, 6);
+  //проверить что все названия вкладок в CS и SS отображаются правильно (нет ошибок типа EvEnTStrEAm и т.д.);
+  //(проверить проход по масиву server side c проверкой по циклам по каждому элемента);
+  const serverSide = await I.grabTextFromAll(loginPage.serverSideLoginContainer);
```
Comment (KaledinaDARIA) [inline CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts:18]: This is unnecessary to create variable only to transport in one place.
```
loginPage.checkElements(await I.grabTextFromAll(loginPage.serverSideLoginContainer), false);
loginPage.checkElements([await I.grabAttributeFrom(loginPage.clientSideLoginContainer, "title")]);
```

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    if (isClientSide) {
+      checkArr = ["iQ Tag Management"];
+    } else {
+      checkArr = ["EventStream", "AudienceStream", "DataAccess", "Predict", "Functions", "Data Connect"];
+    }
+    for (let i: number = 0; i < arr.length; i++) {
+      this.I.assert(checkArr.includes(arr[i]), true, `${arr[i]} is not visible.`);
+    }
+  }
+  public checkErrorMessage(text: string): void {
```
Comment (KaledinaDARIA) [inline CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts:37]: please add to function  this.I.waitForVisible(this.errorMessage, 10);
and remove the same line from main.ts (line 25)

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    if (text) {
```
Comment (KaledinaDARIA) [inline CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts:38]: You only have conditions here and no verification. Please add assert f.e. :
```this.I.assert(text === "Please include a valid email address." || text === "Invalid Password", true, `Invalid text: ${text}`);```

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts (hunk 2 above)
Comment (kate-bobryk) [reply CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts]: Code was updated.

File: CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts (hunk 3 above)
Comment (kate-bobryk) [reply CodeceptJS_PracticalTasks/practical_21_LoginTest/LoginPage.ts:45]: Function was updated, extra line was removed.

**Review by KaledinaDARIA — APPROVED** [review]

**Pull request merged**